regression-tests.api/test_cryptokeys.py

```diff
@@ -0,0 +1,269 @@
+import subprocess
+import json
+import unittest
+
+from test_helper import ApiTestCase, is_auth
+
+@unittest.skipIf(not is_auth(), "Not applicable")
+class Cryptokeys(ApiTestCase):
+    zone = "cryptokeyzone"
+    def setUp(self):
+        super(Cryptokeys, self).setUp()
+        try:
+            subprocess.check_output(["../pdns/pdnsutil", "--config-dir=.", "create-zone", Cryptokeys.zone])
+        except subprocess.CalledProcessError as e:
+            self.fail("Couldn't create zone: "+e.output)
+
+
+    def tearDown(self):
+        super(Cryptokeys,self).tearDown()
+        try:
+            subprocess.check_output(["../pdns/pdnsutil", "--config-dir=.", "delete-zone", Cryptokeys.zone])
+        except subprocess.CalledProcessError as e:
+            self.fail("Couldn't delete zone: "+e.output)
+
+
+    # This methode test the DELETE api call. Sometimes 200 or 422 are valid return codes,
+    # because it "could" happen that the backend crashes during the test or the remove function returns false.
+    # In this case 422 is correct. I don't know how to force false backend behavior. So i assume that the backend
+    # works correct.
+    def test_delete(self):
+        try:
+            keyid = subprocess.check_output(["../pdns/pdnsutil", "--config-dir=.", "add-zone-key", Cryptokeys.zone, "ksk"])
+        except subprocess.CalledProcessError as e:
+            self.fail("pdnsutil add-zone-key failed: "+e.output)
+
+        #checks the status code. I don't know how to test explicit that the backend fail removing a key.
+        r = self.session.delete(self.url("/api/v1/servers/localhost/zones/"+Cryptokeys.zone+"/cryptokeys/"+keyid))
+        self.assertTrue(r.status_code == 200 or r.status_code == 422)
+
+        # Check that the key is actually deleted
```

Comment:
split the test function per each comment that you wrote. if you write comment lines describing what the test code below does, it's a new test case.